tables/mdm/mdm.go

```diff
@@ -160,20 +167,103 @@ func getMDMProfileStatus() (profileStatus, error) {
 	}, nil
 }
 
-func getDEPStatus() (depStatus, error) {
-	cmd := exec.Command("/usr/bin/profiles", "show", "-type", "enrollment")
-	out, err := cmd.Output()
+// Either get the live DEP capability status, or return from the cache if needed.
+func getDEPStatus(status profileStatus) (depStatus, error) {
+	// if we are enrolled via dep, we are by definion dep capable
+	if status.DEPEnrolled {
+		return depStatus{DEPCapable: true}, nil
+	}
+	var depstatus depStatus
+	hasAlreadyChecked := hasCheckedCloudConfigInPast24Hours()
+	if !hasAlreadyChecked {
+		cmd := exec.Command("/usr/bin/profiles", "show", "-type", "enrollment")
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			if strings.Contains(string(out), "Request too soon") {
+				depCapable := getCachedDEPStatus()
+				depstatus.DEPCapable = depCapable
+				return depstatus, nil
+			}
+			return depstatus, nil
+		}
+
+		lines := bytes.Split(out, []byte("\n"))
+
+		if len(lines) > 3 {
+			depstatus.DEPCapable = true
+		}
+	}
+
+	depCapable := getCachedDEPStatus()
+	depstatus.DEPCapable = depCapable
+
+	return depstatus, nil
+}
+
+// Returns true if the device has checked it's cloud config record in the past hour, false if the file is missing or the time is more thab 24 hours ago
+func hasCheckedCloudConfigInPast24Hours() bool {
+	if !fileExists(CloudConfigTimerCheck) {
+		return false
+	}
+
+	var cloudConfigTimerCheck cloudConfigTimerCheck
+	plistFile, err := os.Open(CloudConfigTimerCheck)
 	if err != nil {
-		fmt.Println(err)
+		// could not open file
+		return false
 	}
 
-	lines := bytes.Split(out, []byte("\n"))
+	byteValue, err := ioutil.ReadAll(plistFile)
+	if err != nil {
```

Comment:
It seems bad on the surface to blackhole these errors. We should return them I think?

Reply:
These being errors aren't actually errors - i.e. the program isn't failing, they're just representing a valid state. Also, osquery extensions don't generally log errors, they just do not return anything if there is an issue or they aren't applicable.